internal/extender/demand.go

```diff
@@ -69,25 +69,25 @@ func (s *SparkSchedulerExtender) createDemandForExecutor(ctx context.Context, ex
 			},
 		},
 	}
-	s.createDemand(ctx, executorPod, units)
+	s.createDemand(ctx, executorPod, units, false)
```

Comment:
nit: that ties into what we discussed offline - you have no way of knowing what this `false` does apart from jumping to function definition. Doing Java-style constant like `DONT_ENFORCE_SINGLE_AZ_SCHEDULING` might alleviate that.

non-nit: is this truly what we want? The only codepath that leads to here is re-scheduling executors, but I don't know if that will always be true. At the very least, let's provide a justification.

Reply:
On the nit: Yeah this is true, I ended up doing this way to avoid the proliferation of code paths. There are a few functions which wrap demand creation, we'd end up having duplication of those functions which ended I thought ended up being a little ugly. Happy to re-visit and see if I can reduce duplication here though


Non nit: Yep, we are explicitly not handling executor re-schedules in this iteration for a few reasons
* We do not currently have a way to create demands for a particular AZ http://github.com/palantir/k8s-spark-scheduler-lib/blob/02be893331748732b652e914a50bc20487161677/pkg/apis/scaler/v1alpha2/types_demand.go#L78-L91 
* If support for the above was added, it would need some more work in scheduler to associate the rescheduled executor with the single AZ application
* Metrics suggest that rescheduled executors are created fairly infrequent so we don't have much incentive to invest in this given the above

 
I'll add a comment so that it is clear!

Reply:
Ah yes the constant makes a lot of sense here! Let me do that!